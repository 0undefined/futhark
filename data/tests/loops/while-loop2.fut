-- While-loop with a condition that consumes something that it has allocated itself.
-- ==
-- input {
--   [5,4,2,8,1,9,9]
--   4
-- }
-- output {
--   [5, 4, 2, 8, 6, 9, 9]
-- }

fun pointlessly_consume(x: int, a: *[]int): bool =
  x < reduce((+), 0, a)

fun main(a: *[]int, i: int): []int =
  loop (a) = while pointlessly_consume(a[i], copy(iota(i))) do
    let a[i] = a[i] + 1 in
    a in
  a
