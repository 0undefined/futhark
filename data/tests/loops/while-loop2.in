[5,4,2,8,1,9,9]
4
