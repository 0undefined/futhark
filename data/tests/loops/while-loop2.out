[5, 4, 2, 8, 6, 9, 9]
